torch/csrc/jit/tensorexpr/tensorexpr_tutorical.py

```diff
@@ -0,0 +1,219 @@
+mc = m + 5
+mn = m + n
+e = te.sin(mc * mn)
+print("\nTE arithmetic expression:")
+print(e)
+
+# TE buffer variables
+A = te.BufHandle("A", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+B = te.BufHandle("B", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+print("\nTE buffer variables:")
+print(A, B)
+
+# TE load expression
+i = te.VarHandle("i", torch.int32)
+j = te.VarHandle("j", torch.int32)
+print("\nTE load expression:")
+print(A.load([i, j]))
+
+# TE store statement
+C = te.BufHandle("C", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+store = C.store([i, j], A.load([i, j]) + B.load([i, j]))
+print("\nTE store statement:")
+print(store)
+
+# TE block statement
+block = te.Block([store])
+print("\nTE block statement:")
+print(block)
+
+# TE conditional statement
+cond = te.Cond.make(i > j, block, None)
+print("\nTE cond statement:")
+print(cond)
+
+# TE for statement
+forst = te.For.make(j, te.ExprHandle.int(0), te.ExprHandle.int(32), block)
+forst = te.For.make(i, te.ExprHandle.int(0), te.ExprHandle.int(32), forst)
+print("\nTE for statement:")
+print(forst)
+
+# See more TE IRs: torch/csrc/jit/tensorexpr/IRSpecification.md
+
+# 3. Use TE loop transformations to optimize the computation
+loopnest = te.LoopNest(te.Block([forst]), [C])
+print("\nTE loopnest:")
+print(loopnest.root_stmt())
+
+# obtain the loop handles
+# 1) LoopNest::root_stmt() returns the Block stmt we constructed above; 2)
+# Block::stmts() returns the list of stmts in the block, i.e., [`forst`]; 3) here we
+# use `forst_i` to point to the stmt `forst`.
+forst_i = loopnest.root_stmt().stmts()[0]
+print("\nTE loopnest for-i stmt:")
+print(forst_i)
+# LoopNest::get_loop_at(root, indicies): returns the For stmt indexed by
+# 'indices' in the 'root' For stmt.  Here 'indices' indicates the path to the
+# returned loop from 'root' in AST. `forst_j` is directly embeded in the body
+# of `forst_i` as the only stmt, i.e., the path to `forst_j` from `forst_i`
+# is [0]. We use `get_loop_at` to obtain the handle of `forst_j`.
+forst_j = loopnest.get_loop_at(forst_i, [0])
+print("\nTE loopnest for-j stmt:")
+print(forst_j)
+
+# tile(x, y, x_factor, y_factor): it takes a 2d domain (x, y) and splits the
+# domain into small rectangular blocks each with shape (x_factor, y_factor).
+# The traversal over the domain turns into an outer iteration over the blocks
+# and an inner traversal over all points in the block.
+loopnest.tile(forst_i, forst_j, 8, 8)
+loopnest.simplify()
+print("\nTE loopnest after tiling:")
+print(loopnest.root_stmt())
+
+# vectorize_inner_loops(): find the inner-most loops and vectorize them.
+# Note: this only works for the LLVM backend currently, when no reductions are
+# involved.
+loopnest.vectorize_inner_loops()
+loopnest.simplify()
+print("\nTE loopnest after vectorization:")
+print(loopnest.root_stmt())
+
+# TODO: add more loop transformations
+
+# prepare for codegen
+loopnest.prepare_for_codegen()
```

Comment:
Here would be a good place to show stmt before and after `prepare_for_codegen`, explain what and why we're doing there.